components/Searchbar.js

```diff
@@ -0,0 +1,169 @@
+			return false;
+		} else {
+			return isDeepChildOf(child.parentNode, parent);
+		}
+	}
+	*/
+
+	// event listeners on page load
+	useEffect(() => {
+		window.addEventListener('keydown', (e) => {
+			if (e.key === '/' || (e.key === 'k' && e.ctrlKey)) {
+				if (searchRef.current && e.target !== searchRef.current) {
+					e.preventDefault();
+					searchRef.current.focus();
+				}
+			} else if (e.key === 'Escape') {
+				e.preventDefault();
+				searchRef.current.value = '';
+				searchRef.current.blur();
+				setSearchItems('hidden');
+				setSelected(null);
+				setInputValue('');
+			}
+		});
+	}, []);
+
+	// list of books to be shown under search
+	const [searchItems, setSearchItems] = useState('hidden');
+	// selected book in list
+	const [selected, setSelected] = useState(null);
+	// current value of input
+	const [inputValue, setInputValue] = useState('');
+
+	return (
+		<div className="relative my-auto h-6 w-48">
+			<div
+				className="absolute ml-3 bg-white overflow-y-auto rounded-xl z-20 shadow-2xl"
+				onKeyDown={
+					(e) => {
+						if (searchItems instanceof Array && searchItems.length > 0) {
+
+							if (e.key === 'ArrowUp') {
+								// move selected item up
+
+								e.preventDefault();
+								const idx = (typeof selected !== 'number') ? searchItems.length - 1 : (selected > 0) ? selected - 1 : searchItems.length - 1;
+								setSelected(idx);
+								searchRef.current.value = searchItems[idx].name;
+							} else if (e.key === 'ArrowDown') {
+								// move selected item down
+
+								e.preventDefault();
+								const idx = (typeof selected !== 'number') ? 0 : (selected < searchItems.length - 1) ? selected + 1 : 0;
+								setSelected(idx);
+								searchRef.current.value = searchItems[idx].name;
+							} else if (e.key === 'Enter' && searchItems[selected]) {
+								// navigate to the selected item
+								router.push(`/${searchItems[selected].testament}/${searchItems[selected].name.toLowerCase()}`);
+							}
+						}
+					}
+				}
+			>
+				<input
+					onChange={
+						(e) => {
+							// grab the current value
+							const v = e.target.value.toLowerCase();
+							// save the input value
+							setInputValue(v);
+							if (v.length >= 1) { // there is content
+								// match items
+								let searchedBooks = books.filter(book => book.name.toLowerCase().includes(v.toLowerCase()));
+								// sort by earliest occorence of search first
+								searchedBooks = searchedBooks.sort((first, second) => first.name.toLowerCase().indexOf(v) - second.name.toLowerCase().indexOf(v));
+								// limit the length of the searched items
+								searchedBooks.length = Math.min(searchedBooks.length, Infinity);
+								setSearchItems(searchedBooks);
+							} else {
+								// hide search box
+								setSearchItems('hidden');
+							}
+							setSelected(null);
+						}
+					}
+					onKeyDown={
+						(e) => {
```

Comment:
Again, we need a comment here that explains the function of this block of code.

I can see that it's handling what happens when the enter key is pressed, but then we dive into complex syntax.
Your comment that you have inside:
`// if the current search value is a book name
// navigate to that book`
is great. It's just too keep into the function. It happens after the if, no people are trying to decipher the if statement before then even read the comment that tells them what the if does.

Is what I am saying here making sense?